src/mqtt_client/mqtt_client.rs

```diff
@@ -321,14 +336,28 @@ where
                 }
             };
 
-            log::info!("Settings update: `{}`", path);
-
-            let message: SettingsResponse = settings
-                .string_set(path.split('/').peekable(), message)
-                .map(|_| {
-                    update = true;
-                })
-                .into();
+            let mut buffer = [0; MESSAGE_SIZE];
+            let message: SettingsResponse =
+                match settings.string_get(path.split('/').peekable(), &mut buffer) {
```

Comment:
I'm not super happy with this implementation. It makes us serialize the setting that is being updated on every change. We could alternatively just require that `Settings: Copy` and make a deep copy of the settings structure. It's unknown which one would be the least overhead.

Additionally, the code flow here is pretty convoluted and could likely be broken out into a separate function or refactored.

Reply:
Hmm. Yes that doesn't look good.
1. To separate requested settings from actual active setting on mqtt we may want to separate those trees on the topic level.
2. I would usually expect the sequence to be (1) deser of the payload, (2) validation (3) application/update. Maybe 2 and 3 can both be done by the handler.